### ISSUE TITLE
Disable specific rule(s) in specific file(s)

### ISSUE THREAD

**Issue by avinoamsn** [DavidAnson/vscode-markdownlint#229]
I have a file (`commit.md`) that I use to write git commit messages during development (before committing with the CLI). This file follows all `markdownlint` rules, save for MD041 (requires that the first line is `h1`).

I know that I can disable MD041 with an inline flag (e.g. ``), but this flag will be included in the git message when I commit. What I'd like to do is disable MD041 for `commit.md` from a separate config file, something like:

```JSON
"markdownlint.config": {
	"MD041": { "ignore": ["commit.md"] }
},
```

**OR**

```JSON
"markdownlint.ignore": [
	{
		"commit.md": ["MD041"]
	}
]
```

Looking through the documentation, I don't see a straightforward way to do this. Is a config like this currently possible? Any suggestions or advice is appreciated. Thanks!

**Comment by nschonni** [comment]
Since commit messages use the `#` for a comment line, did you try `# <!-- markdownlint-disable-next-line MD041 -->`, or does that turn it into a heading

**Comment by avinoamsn** [comment]
> Since commit messages use the `#` for a comment line, did you try `# <!-- markdownlint-disable-next-line MD041 -->`, or does that turn it into a heading

@nschonni I had thought of this too, and it in turn violates MD022 (blanks around headings). However, I'm not sure if git includes blank lines at the beginning/end of commit messages, so it may work if I just use a `#` for the first line & add a space b/w that & the commit message's header.

**Comment by DavidAnson** [comment]
Does this work?

https://dlaa.me/markdownlint/#%25mCommit%20message%0A%0AMore%20message%0A%0A%23%20Comment%0A%0A%23%20%3C!--%20markdownlint-configure-file%20%7B%20%22first-line-heading%22%3A%20false%20%7D%20--%3E%0A

**Comment by avinoamsn** [comment]
@DavidAnson thanks for the suggestion! This solution improves on @nschonni's suggestion, as it doesn't introduce any elements to the beginning of the commit message & doesn't raise any resultant linting errors.

Nevertheless, it doesn't solve issue of git not omitting the line when passing in the `commit.md` file (git must not be parsing the file). But using something like `sed '$d' commit.md` to strip out the bottom line before passing it to git is a pretty trivial task. I'll update if I come across a clever solution for this bit.
